pkg/stub/handler.go

```diff
@@ -304,6 +305,43 @@ func (h *Handler) updateCfgArch(cfg *v1.Config) *v1.Config {
 	return cfg
 }
 
+//TODO leaving this in for now to promote discussion of this which started in planning and enhancement proprosal;
+// can delete if need be as PR review process completes 
+func (h *Handler) Disconnected() bool {
+	if h.coreclient == nil {
+		//path possible with unit tests
+		return false
+	}
+	pods, err := h.coreclient.Pods(v1.OperatorNamespace).List(metav1.ListOptions{})
+	if err != nil {
+		logrus.Warningf("Problem reading in pods to determine image and whether disconnected: %s", err.Error())
+		return false
+	}
+	if pods == nil {
+		// path possible with unit tests
+		return false
+	}
+	img := ""
+	for _, pod := range pods.Items {
+		for _, c := range pod.Spec.Containers {
+			img = strings.TrimSpace(c.Image)
+			if strings.HasPrefix(img, "registry.redhat.io") ||
+				strings.HasPrefix(img, "registry.svc.ci.openshift.org") {
+				logrus.Info("Samples operator install payload comes from known, network accessible registry")
+				return false
+			}
+		}
+	}
+	out, err := exec.Command("ping", "registry.redhat.io", "-c 5", "-i 3").Output()
```

Comment:
this is "more valid" but i suspect we'll run into network envs where icmp is blocked.  (or even registry.redhat.io disabling icmp for some reason at some point).

Reply:
better to attempt an actually https connection

Reply:
Fodder for further conversation - I can see a customer asking "why is the Samples operator trying to make a request to registry.redhat.io?"

Reply:
true ... assuming a https connection with skip tls verify is OK ?   ...simplifies things somewhat ...

Reply:
strictly speaking you need to take any proxy config into account, otherwise you're going to skip samples install in a cluster that requires a proxy.  (and samples do work/import even if the cluster requires a proxy).

skipping tls for this check is probably fine which does simplify the proxy-config-handling logic.

Reply:
(if there is not a proxy, you should not need to skiptls though, since the redhat registry should be signed with a CA that's in your system trusts already)

Reply:
Between this and the ICSP thread above, and given the https://github.com/openshift/enhancements/pull/172#discussion_r363490443 comment regarding a simple "we are disconnected" bit in the cluster, thoughts on the feasibility of simply pushing for this instead @bparees ?

Reply:
simply pushing for the explicit config indication?  i'm all for pushing that discussion to a conclusion and then coming back to this if we don't get what we want there.

Reply:
yes that ... do you have suggestions on how we should propagate the discussion past https://github.com/openshift/enhancements/pull/172#discussion_r363490443 ... who/how we should engage @bparees ?

Reply:
seems like we should start with @crawford though i feel like this was pushed back on previously.  Still seems to me that it's information that other operators would find useful.

Reply:
(the other issue may have been that literally nothing actually knows..there's no explicit "this is a disconnected install" setting/config that gets passed in during installation, so it's not clear how the config info would get set in the first place.  That may have been the blocker last time we discussed this)

Reply:
OK, I'm going to circle back then to just determining by network connection.  With that, I'm thinking
- I'll check proxy config ... if one exists, I'm going to punt, and say "not disconnected, or connected".  It seems odd to me if a customer requires a proxy when they are also running disconnected.  Or such a combination is obscure enough that I don't want to deal with getting the proxy related certs.

But if proxy and disconnected together is a thing, please enlighten me.

- With that, I'll try to just make the https connection ... any unexpected items with certs, I'll go skip tls

@bparees thoughts?

- re: customer worried about the samples operator accessing `registry.redhat.io` @adambkaplan ... IMO there will be some much activity to that from the image registry, a small set of connection attempts from the samples operator will get lost in the shuffle

Reply:
your supposition seems reasonable to me (that if they've configured a proxy, that proxy is enabling external access and thus imagestream import should succeed as long as all our import components are properly consuming the proxy).

Reply:
ok @bparees I've pushed the new form that ignores the image for the operator pod, and then
1) sees if a proxy is configured 
2) tries for one mintue to get a successful https / tls connection to registry.redhat.io

PTAL